### PR TITLE
seperation from reader

### PR DESCRIPTION
**Pull request by cgxxv**
I separate nsqio from reader, as the publish or other commands need to write synchronously. Do you have any good ideas?
@mreiferson @ploxiln

### CODE REVIEW

File: nsq.h [hunk 1]
```diff
@@ -15,15 +15,11 @@
 struct Buffer; // from evbuffsock.h
 
 typedef enum {NSQ_FRAME_TYPE_RESPONSE, NSQ_FRAME_TYPE_ERROR, NSQ_FRAME_TYPE_MESSAGE} frame_type;
-typedef enum {NSQ_PARAM_TYPE_INT, NSQ_PARAM_TYPE_CHAR} nsq_cmd_param_type;
+typedef enum {NSQ_PARAM_TYPE_INT, NSQ_PARAM_TYPE_CHAR} nsqCmdParamType;
+typedef enum {NSQ_LOOKUPD_MODE_READ, NSQ_LOOKUPD_MODE_WRITE} nsqLookupdMode;
```
Comment (ploxiln) [inline nsq.h:19]: this is weird, because you should not use lookupd for a nsq writer (publisher), only for a nsq reader (consumer)

**Comment by ploxiln** [comment]
The writer should not really be "synchronous", the user should provide a callback which will be called with notification of success ("OK") or some error. The writer can send multiple PUB before receiving notification of the status of the first, they are "pipelined" (the library just has to keep track of the order of commands to know which the response is for).

I recommend trying to open smaller pull requests, with one step of change at a time (that is still able to build and run).

**Comment by stackerzzq** [comment]
Thanks, I've made a test to run and build in every commit, please take a look again. By the way, I'm sure the nsqio separation from reader is needed. How do you think?

File: nsq.h (hunk 1 above)
Comment (stackerzzq) [reply nsq.h]: I mean to use lookupd to select one nsqd_connection for producer (it seems no need to do this job when the topic_name does not exists at the first time), the consumer also does the same thing. OK, I remove this code firstly.

**Comment by mreiferson** [comment]
Hi @stackerzzq, we just landed #26 — I realize this hasn't been documented anywhere, but it makes it a little easier to make contributions that follow the style of C we prefer by using `make fmt`.

If you wouldn't mind rebasing this, and dropping the `typedef` changes, I'd be happy to review the resulting diff.

Thanks!

**Pull request closed without merging**